client/src/connection_cache.rs

```diff
@@ -480,6 +492,45 @@ impl Default for ConnectionCache {
     }
 }
 
+#[derive(Clone)]
+enum Connection {
```

Comment:
Perhaps ConnectionPool is more appropriate?

Reply:
Sorry, I don't understand what you mean here.  ConnectionPool contains a vec of these Connection types.  Do you mean moving the indirection?  Meaning that we have:

```
struct ConnectionPool {
    connections: Connections
}
enum Connections {
    Udp(Vec<Arc<UdpSocket>>),
    Quic((Vec<Arc<QuicClient>>, Arc<QuicLazyInitializedEndpoint>)),
}
```

Reply:
`ConnectionPool` is already the name of the set of `Connection` -- how about `BaseTpuConnection` since it can either create a UDP or QUIC connection in either blocking or nonblocking mode?